From 4d567852d410ad7b73bf655d40bf8103687780f0 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 28 Dec 2015 17:45:16 +0800
Subject: [PATCH 0804/5242] MLK-12092: mxc_edid: Free mode buffer before
 function return

commit  5b75dc48d1fc676de5d2f35895d4d0109ca5de90 from
https://source.codeaurora.org/external/imx/linux-imx.git

Free mode buffer before function return to avoid memory leak.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_edid.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mxc_edid.c b/drivers/video/fbdev/mxc/mxc_edid.c
index 998b2d4..23110ce 100644
--- a/drivers/video/fbdev/mxc/mxc_edid.c
+++ b/drivers/video/fbdev/mxc/mxc_edid.c
@@ -578,8 +578,10 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 
 	m = kmalloc((num + specs->modedb_len) *
 			sizeof(struct fb_videomode), GFP_KERNEL);
-	if (!m)
+	if (!m) {
+		kfree(mode);
 		return 0;
+	}
 
 	if (specs->modedb_len) {
 		memmove(m, specs->modedb,
-- 
1.7.9.5

